### ISSUE TITLE
[Perf] Browser CPU regression in SafeBrowsing::VerifyChecksum after cr130

### ISSUE THREAD

**Issue by atuchin-m** [brave/brave-browser#41681]
```
Alert information:
Median of segment before: 949.8530000000001
Median of segment after: 1413.136
Relative change: 48.8%
Test: BravePerf/mac-mini-m1-typical/system_health.common_desktop/cpuTime:all_processes:all_threads:all_stages:all_initiators/load_chrome/load_chrome_blank
```

* All the desktop platform are affected
* Chromium builds also have the same issue.
* Chromium revision range: `129.0.6668.89 - 130.0.6723.31`

<img width="50%" src="https://github.com/user-attachments/assets/7bce9fce-396c-4356-ab97-fe6030a67847" />

**Comment by atuchin-m** [comment]
cr129/cr130:
![Image](https://github.com/user-attachments/assets/d522601c-662d-40c8-afb1-427b70637038)

**Comment by atuchin-m** [comment]
An upstream issue with steps to verify: https://crbug.com/373928217

**Comment by atuchin-m** [comment]
The issue is resolved for v1.73.44 for all the affected platforms:
![Image](https://github.com/user-attachments/assets/9a3e552f-30d1-4503-8f8e-fc73171e5630)
